src/main/java/com/dezorganizacja/bridge/view/graphical/FXApplication.java

```diff
@@ -16,14 +23,60 @@ public static void setMainPresenter(MainPresenter mainPresenter) {
 
     @Override
     public void start(Stage stage) throws Exception {
-        FXMLLoader loader = new FXMLLoader(getClass().getResource("/fxml/main.fxml"));
-        Parent root = loader.load();
+        FXApplication.stage = stage;
+        FXApplication.mainPresenter.addObserver(this);
+        loadMainScene();
+    }
+
+    private void loadMainScene() {
+        String programState = mainPresenter.getState();
+        String fxml = stateToFXML(programState);
+
+        if (FXApplication.fxml != null && FXApplication.fxml.equals(fxml)) {
+            return;
+        }
+
+        FXMLLoader loader = new FXMLLoader(getClass().getResource(fxml));
+        Parent root;
+        try {
+            root = loader.load();
+        } catch (IOException e) {
+            System.err.println("error while loading " + fxml + " resource");
+            e.printStackTrace();
+            return;
+        }
+
         stage.setTitle("Bridge");
         stage.setScene(new Scene(root, 300, 275));
 
-        MainController controller = loader.getController();
-        controller.init(mainPresenter);
+        Controller controller = loader.getController();
+        if (controller != null) {
+            controller.init(mainPresenter);
+        }
 
         stage.show();
     }
+
+    public void changeMainScene() {
+        // TODO thread safety
+        Platform.runLater(this::loadMainScene);
+    }
+
+    @Override
+    public void update(Observable observable, Object o) {
+        changeMainScene();
+    }
+
+    private static String stateToFXML(String state) {
```

Comment:
A jeśli będzie Stringiem to wystarczy mapa... Osobiście nie przeszkadza mi typ pola `state`. To i tak jest coś, co będzie do rozkminienia bardziej w przyszłości.